model/src/pyrenew/model/admissionsmodel.py

```diff
@@ -245,46 +258,57 @@ def sample(
             *_,
-        ) = self.sample_latent_admissions(
-            infections=basic_model.latent_infections,
+        ) = self.sample_latent_hosp_admissions(
+            latent_infections=basic_model.latent_infections,
             **kwargs,
         )
-        i0_size = len(latent) - n_timepoints
-        if self.observation_process is None:
-            sampled = None
+        i0_size = len(latent_hosp_admissions) - n_timepoints
+        if self.hosp_admission_obs_process_rv is None:
+            sampled_observed_hosp_admissions = None
         else:
-            if observed_admissions is None:
-                sampled_obs, *_ = self.sample_admissions_process(
-                    predicted=latent,
-                    observed_admissions=observed_admissions,
+            if observed_hosp_admissions is None:
+                (
+                    sampled_observed_hosp_admissions,
+                    *_,
+                ) = self.sample_admissions_process(
+                    observed_hosp_admissions_mean=latent_hosp_admissions,
+                    observed_hosp_admissions=observed_hosp_admissions,
                     **kwargs,
                 )
             else:
-                observed_admissions = au.pad_x_to_match_y(
-                    observed_admissions, latent, jnp.nan, pad_direction="start"
+                observed_hosp_admissions = au.pad_x_to_match_y(
+                    observed_hosp_admissions,
+                    latent_hosp_admissions,
+                    jnp.nan,
+                    pad_direction="start",
                 )
 
-                sampled_obs, *_ = self.sample_admissions_process(
-                    predicted=latent[i0_size + padding :],
-                    observed_admissions=observed_admissions[
+                (
+                    sampled_observed_hosp_admissions,
+                    *_,
+                ) = self.sample_admissions_process(
+                    observed_hosp_admissions_mean=latent_hosp_admissions[
+                        i0_size + padding :
+                    ],
+                    observed_hosp_admissions=observed_hosp_admissions[
                         i0_size + padding :
                     ],
                     **kwargs,
                 )
-            # this is to accommodate the current version of test_model_hosp_no_obs_model. Not sure if we want this behavior
-            if sampled_obs is None:
-                sampled = None
-            else:
-                sampled = au.pad_x_to_match_y(
-                    sampled_obs, latent, jnp.nan, pad_direction="start"
+                # this is to accommodate the current version of test_model_hosp_no_obs_model. Not sure if we want this behavior
+                sampled_observed_hosp_admissions = au.pad_x_to_match_y(
+                    sampled_observed_hosp_admissions,
+                    latent_hosp_admissions,
+                    jnp.nan,
+                    pad_direction="start",
```

Comment:
So, if `sampled_observed_hosp_admissions` is `None`, this works OK by just generating a `nan` array?

Reply:
All tests pass with this code removed, so I will remove it.